gtest/src/program.rs

```diff
@@ -440,28 +440,24 @@ impl<'a> Program<'a> {
         self.id
     }
 
-    pub fn meta_state<E: Encode, D: Decode>(&self, payload: E) -> Result<D> {
-        D::decode(&mut self.meta_state_with_bytes(payload.encode())?.as_slice()).map_err(Into::into)
+    pub fn read_state(&self) -> Result<Vec<u8>> {
+        self.manager.borrow_mut().read_state(&self.id)
     }
 
-    pub fn meta_state_with_bytes(&self, payload: impl AsRef<[u8]>) -> Result<Vec<u8>> {
-        self.manager.borrow_mut().call_meta(
+    pub fn read_state_with_map(
+        &self,
+        mapping_wasm: Vec<u8>,
```

Comment:
I'd provide the method where Wasm is referenced by its path as we do it in `gclient`.